### PR TITLE
feat(core): add locatable interface

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/core/src/locateable/locateable.interface.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/**
+ * An object that has a location referenced by a URI.
+ */
+export interface DaffLocateable {
```
Comment (xelaint) [inline libs/core/src/locateable/locateable.interface.ts:4]: I think it should be `DaffLocatable`

**Review by xelaint — DISMISSED** [review]

File: libs/core/src/locateable/locateable.interface.ts (hunk 1 above)
Comment (griest024) [reply libs/core/src/locateable/locateable.interface.ts]: changed

Comment (damienwebdev) [reply libs/core/src/locateable/locateable.interface.ts]: Agreed.

**Review by damienwebdev — APPROVED** [review]
lgtm

**Pull request merged**